async.go

```diff
@@ -433,26 +421,16 @@ func (c *Async) readLoop() {
 
 		index = 0
 		for index < n {
-
-			if !bytes.Equal(buf[index+protocol.ReservedOffset:index+protocol.ReservedOffset+protocol.ReservedSize], protocol.ReservedBytes) {
-				c.Logger().Error().Err(InvalidBufferContents).Msg("invalid initial bytes in buffer, discarding")
-				break
-			}
-
-			decodedMessage := protocol.Message{
-				From:          binary.BigEndian.Uint32(buf[index+protocol.FromOffset : index+protocol.FromOffset+protocol.FromSize]),
-				To:            binary.BigEndian.Uint32(buf[index+protocol.ToOffset : index+protocol.ToOffset+protocol.ToSize]),
-				Id:            binary.BigEndian.Uint64(buf[index+protocol.IdOffset : index+protocol.IdOffset+protocol.IdSize]),
-				Operation:     binary.BigEndian.Uint32(buf[index+protocol.OperationOffset : index+protocol.OperationOffset+protocol.OperationSize]),
-				ContentLength: binary.BigEndian.Uint64(buf[index+protocol.ContentLengthOffset : index+protocol.ContentLengthOffset+protocol.ContentLengthSize]),
-			}
-
+			p := packet.Get()
```

Comment:
This change lets use use the new message structure while also using the new packet.Get function - which basically 10x our throughput because it doesn't have to keep allocating data over and over again, and instead it can easily reuse buffers